common/utils.py

```diff
@@ -1,25 +1,21 @@
 import os
 import yaml
+from typing import Dict
```

Comment:
Why do we do this instead of just using the builtin `dict` for type hinting, as we do with `str` in other places?

Reply:
For python 3.8 and below, we have to use Dict to indicate the type hint for dictionaries. Python 3.9+ we will be able to use dict, tuple etc. 

Would love a python 3.9 swap in the near future.

Reply:
Go right ahead -- I'm not going to bother with a pbi for that one, as i don't think we'll have any backwards compatibility issues; one of us should just bump it when they wanna.